File: c/recursion/factorial.c
#include <stdio.h>

int factorial(int n);

int main(int argc, char const *argv[])
{
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);
    printf("Factorial is %d.\n", factorial(n));
    return 0;
}

int factorial(int n)
{
    if (n <= 1)
        return 1;

    return n * factorial(n - 1);
}

File: java/no-package/iteration/Factorial.java

import java.util.Scanner;

class Factorial {
    public static void main(String[] args) {
        int fact = 1;
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = scanner.nextInt();
        for(int i=n; i>=1; i--) {
            fact *= i;  // fact = fact * i
        }
        System.out.println("Factorial of " + n + " is " + fact);
    }
}